server/plugin/api_test.go

```diff
@@ -960,3 +961,172 @@ func TestGetUserChannelsForTeam(t *testing.T) {
+				  "approvalId": "mockApproverID",
+				  "organization": "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineRunApprovalRequestStatus: http.StatusInternalServerError,
+			updatePipelineRunApprovalRequestError:  errors.New("not permitted to complete approval"),
+			getRunApprovalDetailsError:             errors.New("failed to approve/reject the request and fetch approval details"),
+			statusCode:                             http.StatusInternalServerError,
+			getRunApprovalDetailsStatus:            http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectRunRequest: invalid payload",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+    			  "approvalId": "mockApproverID",
+    			  "wrong: "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			isPayloadInvalid: true,
+			statusCode:       http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectRunRequest: failed to approve/reject the request due to server error",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": "mockApproverID",
+				  "organization": "mockOrganization",
+				  "projectID": "mockProjectID",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineRunApprovalRequestStatus: http.StatusBadRequest,
+			statusCode:                             http.StatusInternalServerError,
+			updatePipelineRunApprovalRequestError:  errors.New("failed to approve/reject the request due to some internal server error"),
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...)
+			mockAPI.On("GetDirectChannel", testutils.GetMockArgumentsWithType("string", 2)...).Return(&model.Channel{}, nil)
+			mockAPI.On("SendEphemeralPost", mock.AnythingOfType("string"), mock.AnythingOfType("*model.Post")).Return(&model.Post{Message: "mockMessage"})
+			mockAPI.On("UpdateEphemeralPost", mock.AnythingOfType("string"), mock.AnythingOfType("*model.Post")).Return(nil)
+
+			if !testCase.isPayloadInvalid {
+				mockedClient.EXPECT().UpdatePipelineRunApprovalRequest(gomock.Any(), "mockOrganization", "mockProjectID", "test-userID").Return(&serializers.PipelineRunApproveResponse{
+					Value: []*serializers.PipelineRunResponseValue{
+						{},
+					},
+				}, testCase.updatePipelineRunApprovalRequestStatus, testCase.updatePipelineRunApprovalRequestError)
+			}
+
+			if testCase.updatePipelineRunApprovalRequestStatus == http.StatusInternalServerError {
+				mockedClient.EXPECT().GetRunApprovalDetails("mockOrganization", "mockProjectID", "test-userID", "mockApproverID").Return(&serializers.PipelineRunApprovalDetails{}, testCase.getRunApprovalDetailsStatus, testCase.getRunApprovalDetailsError)
```

Comment:
use the constants created in other branches here as well

Reply:
Created a ticket for this. I am doing the same thing across multiple PRs. Once all the PRs are merged, I will create a separate PR for such small enhancements.
https://brightscout.atlassian.net/browse/MI-2511?atlOrigin=eyJpIjoiNWFlOTMxODJjYWM4NDgwYWJiNjE2N2MxMDdjZDRkNjAiLCJwIjoiaiJ9